DataDog/datadog-agent
# [DCA][AD][CLC] Add Prometheus k8s Services Autodiscovery

### What does this PR do?

- Refactor the Prometheus AD logic (pods/services)
- Support Prometheus Autodiscovery using cluster checks

### Motivation

- The DCA should autodiscover prometheus k8s services

### Additional Notes

- Related to https://github.com/DataDog/datadog-agent/pull/6358

### Describe your test plan

- Both DCA and Node Agent must be QA'ed
- Deploy various combinations of the Cluster Agent `prometheus_scrape.checks` config + Prometheus k8s service annotations
- Deploy various combinations of the Node Agent `prometheus_scrape.checks` config + Prometheus pod annotations (pods support wasn't implemented by this PR but the code was changed refactored, let's make sure we didn't break anything)
